### PR TITLE
include service in app.yaml

### LINKED ISSUE #1684: Add service information to the app.yaml
When creating a new flex project add the service information to the app.yaml.

### PR DESCRIPTION
**Pull request by elharo**
fix #1684

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.util/templates/appengine/app.yaml.ftl [hunk 1]
```diff
@@ -0,0 +1,6 @@
+runtime: java
+env: flex
+env_variables:
+  'DBG_ENABLE': 'true'
```
Comment (patflynn) [inline plugins/com.google.cloud.tools.eclipse.util/templates/appengine/app.yaml.ftl]: DBG_ENABLE is no longer required.

File: plugins/com.google.cloud.tools.eclipse.util/templates/appengine/app.yaml.ftl [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+<#if service??>service: ${service}
```
Comment (patflynn) [inline plugins/com.google.cloud.tools.eclipse.util/templates/appengine/app.yaml.ftl]: is service a parameter of the wizard?

File: plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineFlexFacet.java [hunk 3]
```diff
@@ -35,7 +35,7 @@
    * @param project should not be null
    * @return true if project has the App Engine Flex facet and false otherwise
    */
-  public static boolean hasAppEngineFacet(IFacetedProject project) {
+  public static boolean hasFacet(IFacetedProject project) {
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineFlexFacet.java]: There is a counterpart in `AppEngineStandardFacet`, so please rename the method there too.

File: plugins/com.google.cloud.tools.eclipse.util/templates/appengine/app.yaml.ftl (hunk 2 above)
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.util/templates/appengine/app.yaml.ftl]: Yeah, our wizard has an input field for this.

File: plugins/com.google.cloud.tools.eclipse.util.test/src/com/google/cloud/tools/eclipse/util/templates/appengine/AppEngineTemplateUtilityTest.java [hunk 4]
```diff
@@ -143,16 +142,17 @@ private static InputStream getDataFile(String fileName) throws IOException {
     return expectedFileUrl.openStream();
   }
 
-  private static void compareFileContent(
-      InputStream expectedFileStream, InputStream actualFileStream) {
+  private void compareToFile(String expected) throws CoreException, IOException {
+    InputStream testFileStream = testFile.getContents(true);
+    InputStream expectedFileStream = getDataFile(expected);
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.util.test/src/com/google/cloud/tools/eclipse/util/templates/appengine/AppEngineTemplateUtilityTest.java]: What do you think about putting these into the try-resources block?

File: plugins/com.google.cloud.tools.eclipse.util/templates/appengine/app.yaml.ftl (hunk 2 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.util/templates/appengine/app.yaml.ftl]: yes

File: plugins/com.google.cloud.tools.eclipse.appengine.flex.test/src/com/google/cloud/tools/eclipse/appengine/flex/FlexFacetInstallDelegateTest.java [hunk 5]
```diff
@@ -38,13 +43,18 @@
   @Test
   public void testFacetInstall() throws CoreException {
     IProject project = projectCreator.getProject();
+    
+    IProgressMonitor monitor = new NullProgressMonitor();
+    IFolder main = project.getFolder("src/main");
+    ResourceUtils.createFolders(main, monitor);
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.flex.test/src/com/google/cloud/tools/eclipse/appengine/flex/FlexFacetInstallDelegateTest.java]: This test was working before? What made you to add this?

File: plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/FlexFacetInstallDelegate.java [hunk 6]
```diff
@@ -47,15 +48,22 @@ private void createConfigFiles(IProject project, IProgressMonitor monitor) throw
...
     String appYamlParentPath = flexDeployPreferences.getAppEngineDirectory();
     IFolder appYamlParentFolder = project.getFolder(appYamlParentPath);
-    IFile appYaml = appYamlParentFolder.getFile(AppEngineTemplateUtility.APP_YAML);
+    IFile appYaml = appYamlParentFolder.getFile("app.yaml");
     if (appYaml.exists()) {
       return;
     }
+    
+    if (!appYamlParentFolder.exists()) {
+      appYamlParentFolder.create(true, true, subMonitor.newChild(5));
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/FlexFacetInstallDelegate.java]: This works only when the parent of `appYamlParentFolder` exists, so probably better to use `ResourceUtils.createFolders()`.

File: plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineFlexFacet.java (hunk 3 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineFlexFacet.java]: done

File: plugins/com.google.cloud.tools.eclipse.util.test/src/com/google/cloud/tools/eclipse/util/templates/appengine/AppEngineTemplateUtilityTest.java [hunk 7]
```diff
@@ -143,17 +142,16 @@ private static InputStream getDataFile(String fileName) throws IOException {
...
-    String actualContent = actualScanner.useDelimiter("\\Z").next();
-    actualScanner.close();
-
-    Assert.assertEquals(expectedContent, actualContent);
+  private void compareToFile(String expected) throws CoreException, IOException {
+    
+    try (InputStream testFileStream = testFile.getContents(true);
+      InputStream expectedFileStream = getDataFile(expected);
+      Scanner expectedScanner = new Scanner(expectedFileStream);
+      Scanner actualScanner = new Scanner(testFileStream)) {
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.util.test/src/com/google/cloud/tools/eclipse/util/templates/appengine/AppEngineTemplateUtilityTest.java]: Nit: shouldn't these be indented +2 up to this point?

Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.util.test/src/com/google/cloud/tools/eclipse/util/templates/appengine/AppEngineTemplateUtilityTest.java]: I'm not sure. They are independent statements which usually get only +2

Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.util.test/src/com/google/cloud/tools/eclipse/util/templates/appengine/AppEngineTemplateUtilityTest.java]: Looks like the Google formatter uses +4 so I'll take that as authoritative.

**Pull request merged**